x/launch/keeper/msg_server_create_chain.go

```diff
@@ -0,0 +1,65 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/spn/x/launch/types"
+	profiletypes "github.com/tendermint/spn/x/profile/types"
+)
+
+func (k msgServer) CreateChain(goCtx context.Context, msg *types.MsgCreateChain) (*types.MsgCreateChainResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Get the coordinator ID associated to the sender address
+	coordID, found := k.profileKeeper.CoordinatorIdFromAddress(ctx, msg.Coordinator)
+	if !found {
+		return nil, sdkerrors.Wrap(profiletypes.ErrCoordAddressNotFound, msg.Coordinator)
+	}
+
+	// Compute the chain id
+	chainNameCount, found := k.GetChainNameCount(ctx, msg.ChainName)
+	if !found {
+		chainNameCount = types.ChainNameCount{
+			ChainName: msg.ChainName,
+			Count:     0,
+		}
+	}
+	chainID := types.ChainIDFromChainName(msg.ChainName, chainNameCount.Count)
+	chainNameCount.Count++
+
+	// chainID must always be unique by design
+	// if it already exists then something is wrong in the protocol
+	_, found = k.GetChain(ctx, chainID)
+	if found {
+		panic(fmt.Sprintf("chain id %s already exists", chainID))
+	}
+
+	// Initialize the chain
+	chain := types.Chain{
+		ChainID:         chainID,
+		CoordinatorID:   coordID,
+		CreatedAt:       ctx.BlockTime().Unix(),
+		SourceURL:       msg.SourceURL,
+		SourceHash:      msg.SourceHash,
+		LaunchTriggered: false,
+		LaunchTimestamp: 0,
+	}
+
+	// Initialize initial genesis
+	if msg.GenesisURL == "" {
+		chain.InitialGenesis = types.AnyFromDefaultInitialGenesis()
+	} else {
+		chain.InitialGenesis = types.AnyFromGenesisURL(msg.GenesisURL, msg.GenesisHash)
```

Comment:
and we can also use the critical error here to handle invalid genesis

Reply:
I removed those functions because no longer necessary

Also I just put a critical error when `codec.NewAnyWithValue(&types.DefaultInitialGenesis{})` fails but not when `codec.NewAnyWithValue(&types.GenesisURL{...})` fails

DefaultInitialGenesis must succeed because it is a constant and must always be converted into any. So an error is unexpected and critical

For GenesisURL, we have an entry from the message sender (URL and Hash) so it is preferable to just make the message fails because a user must never be able to trigger a critical error